### DISCUSSION TITLE
device.setLocation() on Android

### DISCUSSION THREAD

**Discussion by thuandohoang** [wix/Detox#4463] [Q&A] [answered]
I'm newbie in Android development, in the document of setLocation  method mentions "The correct permissions must be set in your app manifest.". Can someone please explain this point to me? Because the `device.setLocation` method doesn't work on my app. So I suspect there is miss configuration related to the app manifest. Thank you in advance 🙏

**Comment by asafkorem** [comment, marked as answer]
Hey @thuandohoang,
I guess you tried to follow this doc: https://wix.github.io/Detox/docs/api/device/#devicesetlocationlat-lon
It should work, you need to add in your manifest:

```xml
<manifest ...>
   ...
   <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
   ...
   <application ...
```

Do you get any error message when trying to use the `setLocation` method after adding this to your manifest and rebuilding your app?

**Reply by thuandohoang** [reply]
Thank @asafkorem, it's an issue from my side, thanks for your answer.
